book/src/development/emitting_lints.md

```diff
@@ -0,0 +1,217 @@
+# Emitting a lint
+
+Once we have [defined a lint](define_lints.md), written [UI tests](write_tests.md)
+and chosen [the lint pass](lint_passes.md) for the lint, we can begin the
+implementation of the lint logic so that we can emit it and gradually work
+towards a lint that behaves as expected.
+
+Note that we will not go into concrete implementation of a lint logic in this
+chapter. We will go into details in later chapters as well as in two examples
+of real Clippy lints.
+
+In this chapter, we'll see an example of an imaginary `LateLintPass` lint named `bar_expressions`, that checks for
+expresions named `bar`.
+
+To emit a lint with `LateLintPass`, we must implement it for the lint that we have
+declared. Take a look at the [LateLintPass][late_lint_pass] documentation, which
+provides an abundance of methods that we can implement for our lint.
+
+```rust
+pub trait LateLintPass<'tcx>: LintPass {
+    // Trait methods
+}
+```
+
+By far the most common method used for Clippy lints is [`check_expr` method][late_check_expr],
+this is likely because Rust is an expression language and, more often than not,
+the lint we want to work on must examine expressions.
+
+> _Note:_ If you don't fully understand what expressions are in Rust,
+> take a look at the official documentation on [expressions][rust_expressions]
+
+Other common ones include the [`check_fn` method][late_check_fn] and the
+[`check_item` method][late_check_item]. We choose to implement whichever trait
+method based on what we need for the lint at hand.
```

Comment:
The last sentence is a bit too vague. Dropping it completely would not hurt. 

Also the selection of the `check*` function is irrelevant for this chapter (mostly). The only place where it matters is that it influences where an `allow` attribute for the lint can be used and with that for the `*_hir` variants of emitting lints.

Reply:
> Dropping it completely would not hurt.

You're right here, fixed.

---

I disagree in the second part, I find that letting the user know that they can choose to lint in other places doesn't hurt and is useful information.